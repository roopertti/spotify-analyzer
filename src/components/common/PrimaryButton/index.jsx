import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const ButtonBase = styled.button`
  text-transform: uppercase;
  outline: none;
  font-size: 0.9rem;
  font-weight: 500;
  color: white;
  border: none;
  border-radius: 50px;
  background: linear-gradient(90.2deg, #2f29bf -21.41%, rgba(255, 255, 255, 0) 134.41%), #b118c8;
  padding: 1.2rem 4.3rem;
  transition-duration: 0.2s;

  &:hover,
  &:focus {
    box-shadow: 0px 10px 30px rgba(47, 41, 191, 0.3), 0px -10px 30px rgba(177, 24, 200, 0.3);
    transform: translateY(-5px);
  }

  &:active {
    background: linear-gradient(90.2deg, #2f29bf -40%, rgba(255, 255, 255, 0) 90%), #b118c8;
    transform: translateY(0);
    box-shadow: 0px 5px 10px rgba(47, 41, 191, 0.3), 0px -5px 10px rgba(177, 24, 200, 0.3);
  }

  @media (min-width: 960px) {
    font-size: 1rem;
    padding: 1.4rem 5rem;
  }
`;

export default function PrimaryButton({ onClick, children }) {
  return <ButtonBase onClick={onClick}>{children}</ButtonBase>;
}

PrimaryButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};
