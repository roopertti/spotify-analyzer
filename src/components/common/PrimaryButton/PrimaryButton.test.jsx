import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import PrimaryButton from '.';

describe('Button test', () => {
  it('Renders basic button and tests click', () => {
    const mockFn = jest.fn();
    const button = <PrimaryButton onClick={mockFn}>Button</PrimaryButton>;
    const tree = renderer.create(<PrimaryButton onClick={mockFn}>Button</PrimaryButton>).toJSON();
    expect(tree).toMatchSnapshot();
    render(button);
    fireEvent.click(screen.getByText('Button'));
    expect(mockFn).toHaveBeenCalled();
  });
});
