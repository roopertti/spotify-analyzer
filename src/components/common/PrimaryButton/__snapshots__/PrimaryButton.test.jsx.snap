// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Renders basic button and tests click 1`] = `
.c0 {
  text-transform: uppercase;
  outline: none;
  font-size: 0.9rem;
  font-weight: 500;
  color: white;
  border: none;
  border-radius: 50px;
  background: linear-gradient(90.2deg,#2f29bf -21.41%,rgba(255,255,255,0) 134.41%),#b118c8;
  padding: 1.2rem 4.3rem;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c0:hover,
.c0:focus {
  box-shadow: 0px 10px 30px rgba(47,41,191,0.3),0px -10px 30px rgba(177,24,200,0.3);
  -webkit-transform: translateY(-5px);
  -ms-transform: translateY(-5px);
  transform: translateY(-5px);
}

.c0:active {
  background: linear-gradient(90.2deg,#2f29bf -40%,rgba(255,255,255,0) 90%),#b118c8;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  box-shadow: 0px 5px 10px rgba(47,41,191,0.3),0px -5px 10px rgba(177,24,200,0.3);
}

@media (min-width:960px) {
  .c0 {
    font-size: 1rem;
    padding: 1.4rem 5rem;
  }
}

<button
  className="c0"
  onClick={[MockFunction]}
>
  Button
</button>
`;
